src/type_checker/mod.rs

```diff
@@ -392,7 +392,8 @@ impl<B: Backend> TypeChecker<B> {
                             qualified,
                             FnInfo {
                                 is_hint: true,
-                                kind: FnKind::BuiltIn(function.sig.clone(), fn_handle),
+                                // todo: is there a case where we want to ignore argument types for hint functions?
+                                kind: FnKind::BuiltIn(function.sig.clone(), fn_handle, false),
```

Comment:
I couldn't immediately decide what to do here since we first convert a hint function to a builtin one. It felt like we would never disable argument type checking for a hint function so I defaulted to false.

Reply:
The builtin hints will be deprecated soon. So don't worry bout this. 
Good insight!